### ISSUE TITLE
new runner in refine: ExtractIdiomsRunner

### ISSUE THREAD

**Issue by eubinecto** [eubinecto/youtora#160]
## why?
For parsing `IdiomRaw` -> `Idiom`.

**Comment by eubinecto** [comment]
later: extracting example sentences.

**Comment by eubinecto** [comment]
## completed parsing
some example results:
```
### pink slip ####
[   {   'defs': [   {   'context': 'idiomatic, metonymically, US',
                        'examples': None,
                        'pos': 'noun',
                        'text': 'Notice of termination of employment.'},
                    {   'context': 'US',
                        'examples': None,
                        'pos': 'noun',
                        'text': 'The title for an automobile.'},
                    {   'context': 'Australia, New South Wales',
                        'examples': None,
                        'pos': 'noun',
                        'text': 'An automobile roadworthiness inspection '
                                'certificate.'},
                    {   'context': 'US, informal, transitive',
                        'examples': None,
                        'pos': 'verb',
                        'text': 'To terminate the employment of.'}],
        'etymology': 'From pink + slip (“piece of paper”), referring to the '
                     'colour of the paper, in some cases as it was in past, '
                     'and not necessarily present, practice.\n'
                     '(automobile title): Most likely from California, where '
                     'the title is printed on a smaller index-card sized piece '
                     'of pink paper; in other states the title is typically a '
                     'full-size 8 1/2" x 11" sheet, usually printed on blue or '
                     'blue-green paper with borders, similar to a stock '
                     'certificate.\n'}]
### play the field ####
[   {   'defs': [   {   'context': 'sports',
                        'examples': None,
                        'pos': 'verb',
                        'text': 'This term needs a definition. Please help out '
                                'and add a definition, then remove the text '
                                '{{rfdef}}.'},
                    {   'context': None,
                        'examples': None,
                        'pos': 'verb',
                        'text': 'To date more than one person at the same '
                                'time.'},
                    {   'context': None,
                        'examples': None,
                        'pos': 'verb',
                        'text': 'To seek sexual variety.'},
                    {   'context': None,
                        'examples': None,
                        'pos': 'verb',
                        'text': 'To experiment sexually with multiple '
                                'partners.'},
                    {   'context': None,
                        'examples': None,
                        'pos': 'verb',
                        'text': 'To engage in extramarital affairs.'}],
        'etymology': None}]
### one-off ####
[   {   'defs': [   {   'context': None,
                        'examples': None,
                        'pos': 'adjective',
                        'text': 'Occurring once, one-time, independent of any '
                                'pattern.'},
                    {   'context': None,
                        'examples': None,
                        'pos': 'adjective',
                        'text': 'Singular; unique; special; remarkable.'},
                    {   'context': 'idiomatic',
                        'examples': None,
                        'pos': 'noun',
                        'text': 'Something that is done, created, etc. only '
                                'once, and often quickly, simply, or '
                                'improvisationally.'},
                    {   'context': None,
                        'examples': None,
                        'pos': 'noun',
                        'text': 'Something unique and remarkable.'}],
        'etymology': 'From one +\u200e off. Probably from foundry work, for '
                     'which making reusable molds is expensive and expediency '
                     'the rule for molds not to be used only once.\n'
                     'A term long used by artists who are printmakers and '
                     'sculptors to indicate a unique print or casting. If '
                     'using traditional edition numbering, a one-off would be '
                     'a "1/1", which is said, "one of one" meaning the first '
                     'print in an edition of one.\n'}]
### over one's head ####
[   {   'defs': [   {   'context': None,
                        'examples': None,
                        'pos': 'adjective',
                        'text': 'Used other than with a figurative or '
                                'idiomatic meaning: see over,\u200e head.'},
                    {   'context': 'idiomatic',
                        'examples': None,
                        'pos': 'adjective',
                        'text': 'More complex or confusing than one can '
                                'understand; beyond one’s comprehension.'},
                    {   'context': 'idiomatic) More than one can handle; too '
                                   "much (especially in over one's head",
                        'examples': None,
                        'pos': 'adjective',
                        'text': '.'},
                    {   'context': 'idiomatic, sports',
                        'examples': None,
                        'pos': 'adjective',
                        'text': 'Performing at a level greatly superior to '
                                "one's usual level of performance."},
                    {   'context': 'idiomatic, business',
                        'examples': None,
                        'pos': 'adjective',
                        'text': 'Directed to someone with authority over the '
                                'person, so as to avoid requiring the approval '
                                'of the person himself or herself.'}],
        'etymology': None}]
### pick out ####
[   {   'defs': [   {   'context': None,
                        'examples': None,
                        'pos': 'verb',
                        'text': 'to remove by picking'},
                    {   'context': None,
                        'examples': None,
                        'pos': 'verb',
                        'text': 'to select'},
                    {   'context': 'idiomatic',
                        'examples': None,
                        'pos': 'verb',
                        'text': 'to distinguish; discern'},
                    {   'context': None,
                        'examples': None,
                        'pos': 'verb',
                        'text': 'To ornament or relieve with lines etc. of a '
                                'different, usually lighter, colour.'},
                    {   'context': "idiomatic) to detect using one's senses "
                                   '(sight, smell, hearing, touch, taste',
                        'examples': None,
                        'pos': 'verb',
                        'text': "to detect using one's senses (sight, smell, "
                                'hearing, touch, taste)'},
                    {   'context': 'idiomatic, soccer',
                        'examples': None,
                        'pos': 'verb',
                        'text': 'to send a long pass or cross to.'}],
        'etymology': None}]
### pink slime ####
[   {   'defs': [   {   'context': 'paper manufacture',
                        'examples': None,
                        'pos': 'noun',
                        'text': 'An undesirable pink-colored microbial mass '
                                'occurring in the slurry used in making paper. '
                                '[from 1951]'},
                    {   'context': 'informal, dysphemistic',
                        'examples': None,
                        'pos': 'noun',
                        'text': 'A meat byproduct produced from scraps by '
                                'heating and then treating with ammonia to '
                                'produce a food additive. [from 2002]'},
                    {   'context': None,
                        'examples': None,
                        'pos': 'noun',
                        'text': 'Used other than with a figurative or '
                                'idiomatic meaning: see pink,\u200e slime. '
                                'Slime which is pink.'}],
        'etymology': 'Apparently first applied to food in 2002 by Food Safety '
                     'and Inspection Service microbiologist Gerald '
                     'Zirnstein.\n'}]
### public comment ####
[   {   'defs': [   {   'context': 'politics, law) Input given by the public '
                                   'to governmental (or rarely other) bodies, '
                                   'about proposed legislation or regulation(s',
                        'examples': None,
                        'pos': 'noun',
                        'text': ', during a period—and by means—set aside and '
                                'prescribed by law.'},
                    {   'context': None,
                        'examples': None,
                        'pos': 'noun',
                        'text': 'Used other than with a figurative or '
                                'idiomatic meaning: see public,\u200e '
                                'comment.'}],
        'etymology': 'public +\u200e comment'}]
### pull the plug ####
[   {   'defs': [   {   'context': 'intransitive',
                        'examples': None,
                        'pos': 'verb',
                        'text': 'to unplug or cut power'},
                    {   'context': 'intransitive, idiomatic',
                        'examples': None,
                        'pos': 'verb',
                        'text': 'to cease to support; to halt'},
                    {   'context': 'intransitive',
                        'examples': None,
                        'pos': 'verb',
                        'text': 'to cease or turn off life support'},
                    {   'context': 'intransitive, idiomatic',
                        'examples': None,
                        'pos': 'verb',
                        'text': 'to cease from production or publication.'}],
        'etymology': None}]
### put back ####
[   {   'defs': [   {   'context': 'transitive',
                        'examples': None,
                        'pos': 'verb',
                        'text': 'To return something to its original place.'},
                    {   'context': 'intransitive, nautical',
                        'examples': None,
                        'pos': 'verb',
                        'text': 'To turn back; to return.'},
                    {   'context': 'transitive',
                        'examples': None,
                        'pos': 'verb',
                        'text': 'To postpone an arranged event or '
                                'appointment.'},
                    {   'context': 'transitive, idiomatic',
                        'examples': None,
                        'pos': 'verb',
                        'text': 'To drink fast; to knock down alcohol.'},
                    {   'context': 'transitive, idiomatic',
                        'examples': None,
                        'pos': 'verb',
                        'text': 'To change the time in a time zone to an '
                                'earlier time.'}],
        'etymology': None}]


```

**Comment by eubinecto** [comment]
later: also try parsing `alternative_forms`
